examples/Carousel.onlyDots.tsx

```diff
@@ -46,8 +46,7 @@ export default () => {
         {gradients.map(({ from, to }, index) => (
           <li
             key={index}
-            role='tabpanel'
-            id={`${id}-slide-${index}`}
+            id={`${id}--slide-${index}`}
```

Comment:
why was this id changed?

Reply:
There was originally a typo that caused one of the examples to fail, but I should change this back to `-slide-` and then change the example id to match that.

Reply:
There was a typo in the example that caused it to throw an error. It was supposed to be `--slide-`, which is the format used in the `CarouselDotsList` component. 
![image](https://github.com/iTwin/iTwinUI/assets/44583237/87cbc45f-5d0e-44ca-b2de-1a189ec06862)

Reply:
hmm i think this behavior is actually wrong. `CarouselDotsList` is the component, whereas `Carousel.onlyDots` is the example; the latter should not rely on implementation details of the former.

i wonder if we even need `id` and `aria-controls`. since we removed `role=tabpanel` in https://github.com/iTwin/iTwinUI/pull/1416#discussion_r1268630982, i think `role=tab` can also be removed from `CarouselDot`. i will double check tomorrow

Reply:
I removed `role=tab` from `CarouselDot` myself, and it generated these violations. I think we have to  keep this role, unless something else can be changed that would allow us to remove it. 
![image](https://github.com/iTwin/iTwinUI/assets/44583237/fee9f7d3-eed5-47fb-8c57-1b1ca5b15f31)

Reply:
so i've been looking into this and need more time. I think fixing Carousel accessibility would require some bigger changes across the whole component, so you could move this PR to draft and work on a different PR in the meantime.

i will pick up Carousel later